pennylane/workflow/execution.py

```diff
@@ -377,6 +377,7 @@ def execute(
     max_expansion=10,
     device_batch_transform=True,
     device_vjp=False,
+    mcm_config=None,

```

Comment:
How do we resolve `config.mcm_config` versus `mcm_config`? Which one would get priority if both are provided?

Reply:
In `QNode`, `execution_kwargs.mcm_config` is prioritized. In `execute`, `config.mcm_config` is prioritized. This kwarg is provided here because for old devices, `config` is `None`, and in that case we need the `mcm_config` to initialize the new execution config.

Reply:
This is now updated. All validation happens in `QNode.__init__` or `QNode._execution_component` (called by `QNode.__call__`). I've kept the `mcm_config` arg in `qml.execute` though for futureproofing.